Sources/ComposableArchitecture/Reducer/Reducers/Presentation.swift

```diff
@@ -222,7 +222,13 @@ public struct _PresentationReducer<
     case let (.some(destinationState), .some(.presented(destinationAction))):
       let id = self.id(for: destinationState)
       destinationEffects = self.destination
-        .dependency(\.dismiss, DismissEffect { Task.cancel(id: DismissID()) })
+        .dependency(\.dismiss, withEscapedDependencies { escaped in
```

Comment:
Without this, the navigation ID is over-applied when re-entrant, _e.g._:

```diff
-[\.$child, Child.self]
+[\.$child, Child.self, \.$child, Child.self]
```

We should audit for other spots where this might happen...